.travis.yml

```diff
@@ -0,0 +1,40 @@
+language: c
+sudo: required
+env:
+  global:
+    # Travis limits maximum log size, we have to cut tests output 
+    - ZFS_TEST_TRAVIS_LOG_MAX_LENGTH=400
+  matrix:
+    # tags are in ascending order
+    - ZFS_TEST_TAGS='acl,atime,bootfs,cache,cachefile,casenorm,chattr,checksum,clean_mirror,compression,ctime,delegate,devices,events,exec,fault,features,grow_pool,grow_replicas,zdb,zfs,zfs_bookmark,zfs_change-key,zfs_clone,zfs_copies,zfs_create,zfs_diff,zfs_get,zfs_inherit,zfs_load-key'
+    - ZFS_TEST_TAGS='zfs_mount,zfs_promote,zfs_property,zfs_receive,zfs_reservation,zfs_send,zfs_set,zfs_share,zfs_snapshot,zfs_unload-key,zfs_unmount,zfs_unshare,zfs_upgrade,zpool,zpool_add,zpool_attach,zpool_clear,zpool_create,zpool_destroy,zpool_detach'
+    - ZFS_TEST_TAGS='zpool_events,zpool_expand,zpool_export,zpool_get,zpool_history,zpool_import,zpool_labelclear,zpool_offline,zpool_online,zpool_remove,zpool_reopen,zpool_scrub,zpool_set,zpool_status,zpool_sync'
+    - ZFS_TEST_TAGS='inuse,large_files,largest_pool,link_count,migration,mmap,mmp,mount,mv_files,nestedfs,no_space,nopwrite,online_offline,pool_names,poolversion,privilege,quota,raidz,redundancy,rsend'
+    - ZFS_TEST_TAGS='refquota,refreserv,rename_dirs,replacement,reservation,rootpool,scrub_mirror,slog,snapshot,snapused,sparse,threadsappend,tmpfile,truncate,upgrade,userquota,vdev_zaps,write_dirs,xattr,zvol,libzfs'
+    # this builder tags are not in ascending order, they are the manually picked longest ones
+    - ZFS_TEST_TAGS='history,hkdf,inheritance,cli_user,zfs_destroy,zfs_rename,zfs_rollback,zpool_replace,zpool_upgrade'
+before_install:
+    - sudo apt-get -qq update
+    - sudo apt-get install --yes -qq build-essential autoconf libtool gawk alien fakeroot linux-headers-$(uname -r)
+    - sudo apt-get install --yes -qq zlib1g-dev uuid-dev libattr1-dev libblkid-dev libselinux-dev libudev-dev libssl-dev
+    # packages for tests
+    - sudo apt-get install --yes -qq libdevmapper-dev parted lsscsi ksh attr acl nfs-kernel-server fio
```

Comment:
You can remove `libdevmapper-dev` this dependency was only added briefly.  I'll make sure it get's dropped from the wiki documentation.

Reply:
You don't forcibly need to issue arbitrary APT commands, Travis has a module so you can list what packages you need and it will install automatically: https://docs.travis-ci.com/user/installing-dependencies/#Installing-Packages-with-the-APT-Addon

Reply:
@behlendorf thanks, removed it.

@sylveon thanks, unfortunately it doesn't give any gain, only more verbose, which we don't need. I'll revert it.